### ISSUE TITLE
Xcode 11.2b1: Crashes with `EXC_I386_GPFLT` when using `after(seconds: TimeInterval)`

### ISSUE THREAD

**Issue by maniramezan** [mxcl/PromiseKit#1099]
* Using latest stable PromiseKit (6.11.0) through SwiftPM
* This started happening in Xcode 11.2 beta 1
* You can reproduce it by calling `after(seconds: TimeInterval)` anywhere in the code, example:
```
after(seconds: 2).done {
    print("yay")
}
```
* The exception is happening in [Guarantee.swift](https://github.com/mxcl/PromiseKit/blob/master/Sources/Guarantee.swift#L69) `EXC_BAD_ACCESS (code=EXC_I386_GPFLT)`

**Comment by mxcl** [comment]
Report to Apple. Not much we can do about it.

**Comment by mani-ramezan-zocdoc** [comment]
Opened [SR-11567](https://bugs.swift.org/browse/SR-11567) and attached a sample project to it. Weirdly, if I try this in a console app it doesn't crash:

```
after(seconds: 2).done {
    print("Yaaay, no crahs!")
}

RunLoop.main.run()
```

**Comment by maniramezan** [comment]
Opened [SR-11567](https://bugs.swift.org/browse/SR-11567) and attached a sample project to it. Weirdly, if I try this in a console app it doesn't crash:

```
after(seconds: 2).done {
    print("Yaaay, no crahs!")
}

RunLoop.main.run()
```

**Comment by mxcl** [comment]
More details in #1102

**Comment by mxcl** [comment]
I have looked and it is not a bug in PromiseKit as far as I can tell. Swift itself seems the problem. So no, no luck in fixing this, please +1 the ticket at the Swift bug tracker.

**Comment by lgaches** [comment]
Still happening with the Xcode 11.2 release

**Comment by maniramezan** [comment]
Yes, and now I'm thinking for how long Apple would accept apps built with Xcode 11.1. It'll be problematic if they stop accepting 11.1 builds and 11.2 builds all crashes on this 🤷🏻‍♂️

**Comment by ldiqual** [comment]
This is pretty bad, I was really hoping Apple would fix it in the official 11.2 release. @mxcl from what you saw while investigating, is there even a remote chance that we can workaround it, even if a bit hacky?

**Comment by onevcat** [comment]
It seems like an SPM specified problem. As long as I can say, using CocoaPods or Carthage will not trigger this crash.

**Comment by lgaches** [comment]
I can confirm it's related to SPM. Using another method to integrate PromiseKit fixed the problem.

**Comment by tcldr** [comment]
I believe this is caused by SR-11564. (https://bugs.swift.org/browse/SR-11564)

**Comment by ydnar** [comment]
We found a workaround. Setting DEAD_CODE_STRIPPING = NO in our app’s target fixes the crash.

**Comment by cdstamper** [comment]
Any word on when we'll be seeing a fix for this released? @kuglee @mxcl

**Comment by mxcl** [comment]
As already stated multiple times, it's not fixable from PromiseKit itself.

**Comment by onevcat** [comment]
It turns out Xcode 11.3 fixed this issue (although still in beta).

**Comment by danqing** [comment]
11.3 is out of beta now.

**Comment by lucasmpaim** [comment]
Solved when upgraded xcode to 11.3

**Comment by maniramezan** [comment]
Going to close this as it's confirmed that it's fixed in Xcode 11.3.
